swiftgen-cli/ArgumentsUtils.swift

```diff
@@ -47,12 +47,20 @@ enum OutputDestination: ArgumentConvertible {
     }
   }
 
-  func write(content: String) {
+  func writeIfChanged(content: String) {
+    write(content, onlyIfChanged: true)
+  }
+  
+  func write(content: String, onlyIfChanged: Bool = false) {
     switch self {
     case .Console:
       print(content)
     case .File(let path):
       do {
+        if try path.isContentEqualTo(content) && onlyIfChanged {
```

Comment:
It seems like putting the try in front of the whole expression might work.

What about this:

```
if try onlyIfChanged && path.read(NSUTF8StringEncoding) == "Foo" {
  return print("Not writing the file as content is unchanged")
}
```

Given that the method name `isContentEqualTo` doesn't sound very Swifty to me anyway, this would solve both problems :wink: